internal/blobstore/blobstore.go

```diff
@@ -76,22 +76,65 @@ func NewContentChallengeResponse(chal *ContentChallenge, r io.ReadSeeker) (*Cont
 	}, nil
 }
 
+// Store stores data blobs in the configured store.
+type Store struct {
+	store
+}
+
 // Store stores data blobs in mongodb, de-duplicating by
 // blob hash.
-type Store struct {
+type store interface {
+	Put(r io.Reader, name string, size int64, hash string, proof *ContentChallengeResponse) (*ContentChallenge, error)
+	PutUnchallenged(r io.Reader, name string, size int64, hash string) error
+	Open(name string) (ReadSeekCloser, int64, error)
+	Remove(name string) error
+}
+
+// gridStore stores data blobs in mongodb gridfs, de-duplicating by
```

Comment:
I like the type per file and filename matches the type convention, but we haven't been doing that AFAICT.